rosetta-2021.html

```diff
@@ -0,0 +1,236 @@
+<script>
+  // Rule of thumb: put a semicolon at the end of every line that doesn't end in a curly
+
+  // Simple output
+  // =============
+
+  // puts "howdy!"
+
+  // alert("howdy!");
+
+
+  // Declaring local variables
+  // =========================
+
+  // name = "Raghu"
+  // puts "howdy, #{name}!"
+
+  // var name = "Raghu";
+  // alert("howdy, " + name + "!");
+
+  // Note: there's no String interpolation in JS
+
+
+  // Defining and calling methods
+  // ============================
+
+  // def greet
+  //   puts "howdy!"
+  // end
+
+  // greet
+
+  // function greet() {
+  //   alert("howdy!");
+  // }
```

Comment:
There's also [Arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#comparing_traditional_functions_to_arrow_functions), which may or may not be worth mentioning. Again part of ES6, but used pretty often in React. There are not _exactly_ the same since I think `this` (or `self` in Ruby, which might be worth mentioning) is different inside an Arrow function than a regular function.

```js
// Traditional Function
function (a){
  return a + 100;
}

// Arrow Function Break Down

// 1. Remove the word "function" and place arrow between the argument and opening body bracket
(a) => {
  return a + 100;
}

// 2. Remove the body brackets and word "return" -- the return is implied.
(a) => a + 100;

// 3. Remove the argument parentheses
a => a + 100;
```

Reply:
They are especially used when functions take an [anonymous function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions#the_function_expression_function_expression) as an argument.

Reply:
Huh. Neato, I suppose. This only works for single-line functions, I presume?

Reply:
They can be multi-line, but if they are then you need to keep the brackets and `return`.

```js
// Traditional Function
function (a, b){
  let chuck = 42;
  return a + b + chuck;
}

// Arrow Function
(a, b) => {
  let chuck = 42;
  return a + b + chuck;
}
```